taskt/Core/Automation/Commands/Secure Data Commands/SetSecureStringCommand.cs

```diff
@@ -0,0 +1,88 @@
+    public class SetSecureStringCommand : ScriptCommand
+    {
+        [XmlAttribute]
+        [PropertyDescription("Input Variable")]
+        [InputSpecification("Provide the Secure String Variable.")]
+        [SampleUsage("{vSecureString}")]
+        [Remarks("You should provide only the variable containing secure string.")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        public string v_Input { get; set; }
+
+        [XmlAttribute]
+        [PropertyDescription("Output Secure String Variable")]
+        [InputSpecification("Select or provide a variable from the variable list.")]
+        [SampleUsage("vUserVariable")]
+        [Remarks("If you have enabled the setting **Create Missing Variables at Runtime** then you are not required" +
+                  " to pre-define your variables; however, it is highly recommended.")]
+        public string v_OutputUserVariableName { get; set; }
+        public SetSecureStringCommand()
+        {
+            CommandName = "SetSecureStringCommand";
+            SelectionName = "Set Secure String";
+            CommandEnabled = true;
+            CustomRendering = true;
+        }
+
+        public override void RunCommand(object sender)
+        {
+            var engine = (AutomationEngineInstance)sender;
+            var sourceVariable = VariableMethods.LookupVariable(engine, v_Input);
+
+            if (sourceVariable.VariableValue is SecureString)
+            {
+                var targetVariable = VariableMethods.LookupVariable(engine, v_OutputUserVariableName);
+
+                if (targetVariable != null)
+                {
+                    targetVariable.VariableValue = sourceVariable.VariableValue;
+                    targetVariable.IsSecureString = true;
+                }
+                else
+                    engine.VariableList.Add(new ScriptVariable()
+                    {
+                        VariableName = v_OutputUserVariableName.ConvertToUserVariable(engine),
+                        VariableValue = sourceVariable.VariableValue,
+                        IsSecureString = true
+                    });
+            }
+            else
+                throw new Exception("Provided Input Variable for 'SetSecureStringCommand' is not a 'SecureString'.");
+        }
+
+        public override List<Control> Render(frmCommandEditor editor)
+        {
+            //custom rendering
+            base.Render(editor);
+
+            //create control for variable name
+            RenderedControls.AddRange(CommandControls.CreateDefaultInputGroupFor("v_Input", this, editor));
+            RenderedControls.AddRange(CommandControls.CreateDefaultOutputGroupFor("v_OutputUserVariableName", this, editor));
+
+            return RenderedControls;
+        }
+
+        public override string GetDisplayValue()
+        {
+            return base.GetDisplayValue() + $" [Set '{v_Input}' to Variable '{v_OutputUserVariableName}']";
```

Comment:
No we can't use SetVariableCommand for assigning a secure string to a variable because SetVariableCommand facilitates to provide expression like {vNum}+1. Here we just need to assign a secure string.

Reply:
Ok, but the only way to assign a secure string to a variable is to already have one in the first place. So you're just renaming the variable. And yes, SetVariable facilitates mathematical expressions, but it also reassigns variables. So while adding to a secure string would throw an error, renaming it wouldn't. Have you tried using it for that purpose?

Reply:
made the suggested change

Reply:
When assigning a SecureString to another variable, I am not renaming the variable rather I am assigning the SecureString value to another variable. SetVariableCommand works only for String Type Variables and doesn't assign a complex variable value to another variable. Like a DataTable cannot be assigned to another variable using SetVariableCommand.

Reply:
Right, and that should be fixed in my opinion. Set Variable should be able to assign a complex variable to a different one and I didn't realize until now that it can't. It should be able to reassign any complex variable including secure strings. Instead of having multiple set variable commands, we should have one that works for all complex variable.

Reply:
Removed SetSecureStringCommand and fixed Helper method for SetVariableCommand to make it work for all data types.